src/material/tabs/paginated-tab-header.ts

```diff
@@ -246,6 +246,35 @@ export abstract class MatPaginatedTabHeader
     });
   }
 
+  /** A method responsible for sending any change that could affect layout about
+   * items to subscribers.
+   */
+  _getItemChanges() {
+    const itemContainerItemChanged = new Subject<void>();
+
+    // Check to see if ResizeObserver is supported and if not, stub the function
+    // out so that it does nothing if not supported
+    const observer =
+      typeof ResizeObserver === 'function'
+        ? new ResizeObserver(() => {
+            itemContainerItemChanged.next();
+          })
+        : {
```

Comment:
I don't think that we should be stubbing out an observable. You can use the `EMPTY` observable from rxjs instead.

Reply:
We're not. We're stubbing out the RO function. I could also return `null` and check for `null` down below. I don't think EMPTY works.

Reply:
Hmm, I see. I find this side-effect-ey use of `map` confusing. Here's how I would've implemented it:

1. Keep the `merge(dirChange, resize, this._items.changes)` above as is.
2. Have a utility that looks like this:
```
private _itemsResized(): Observable<void> {
  if (typeof ResizeObserver !== 'function') {
    return EMPTY;
  }

  return this._items.changes.pipe(
    startWith(this._items),
    switchMap(
      (items: QueryList<MatPaginatedTabHeaderItem>) =>
        new Observable((observer: Observer<void>) =>
          this._ngZone.runOutsideAngular(() => {
            const resizeObserver = new ResizeObserver(() => observer.next());
            items.forEach(item => resizeObserver.observe(item.elementRef.nativeElement));
            return () => resizeObserver.disconnect();
          }),
        ),
    ),
    // Skip the first emit since the resize observer emits when an item is observed.
    skip(1),
  );
}
```
3. Change the `merge` call above to `merge(dirChange, resize, this._items.changes, this._itemsResized())`.


This approach fixes the following issues:
1. You generally don't want side effects from your `map` operator.
2. The `ResizeObserver` wasn't being disconnected when the subscription is dropped.
3. It avoids the extra call when the elements are observed for the first time.
4. The `ResizeObserver` itself wasn't being created outside the `NgZone`.

Note that this is slightly more wasteful when it comes to recreating resize observers, but I think that it's not a big deal in this case since tabs don't tend to change very often. If it becomes a problem, we can start keeping track of which items are being observed.

Reply:
You can reintroduce the first render by removing the `skip(1)`. I added it, because the first emit seems redundant to me. It fires when the elements are observed for the first time, but the initial render should already be correct since we do it right after the elements are inserted.

Reply:
Removing `skip` will fire it on the RO layout which causes the rest to fail because as it turns out, the `fakeAsyncZone` doesn't actually capture the `ResizeObserver` tick (even though it technically runs before the `paint`, it's not before the call stack is cleared). This explains why that test passed and was not flaky.

We could remove the test and not bother or perhaps try to a stability check hack to fix that if you don't think the first synchronous emission is needed.

Reply:
Okay, that explains why the test was passing. I'd rather avoid the test than have it fire on init, because it has the potential to cause layout thrashing.

Reply:
OK, I removed the test for now to keep it simple especially since MDC is coming "soon". I talked with some internal folks and these links should be OK to share:

Before: https://screencast.googleplex.com/cast/NTA3MjkzNjU0MjE0MjQ2NHw0NTU1YzBmNi1lOA
After: https://screencast.googleplex.com/cast/NjI3MjI0NjU2Njg3OTIzMnw0MjBiMDcxZS0zNQ

This shows the stuff working in lieu of some tests.

Reply:
Shouldn't we get rid of the `startWith(undefined)` as well?

Reply:
It should be gone. I hadn't realize that pushing would immediately send it for re-review, unlike some other tools. :)